### PR TITLE
small vcx.h cleanup, add migration guide

### PR DESCRIPTION
**Pull request by glowkey**

### CODE REVIEW

File: vcx/docs/migration.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## Changelog
+
+-   Schema IDs and credential definition IDs have changed (see libindy migrating guides here: https://github.com/hyperledger/indy-sdk/tree/master/doc).  
+-   The vcx_claimdef_* family of functions is now the vcx_credential_* family.
+-   vcx_credential_* - family of functions for receiving credential offers, sending credential requests and storing issued credentials in the wallet.
```
Comment (hadleym) [inline vcx/docs/migration.md]: What are we trying to say on this line? Can we just combine with line 8?

File: vcx/docs/migration.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Changelog
+
+-   Schema IDs and credential definition IDs have changed (see libindy migrating guides here: https://github.com/hyperledger/indy-sdk/tree/master/doc).  
+-   The vcx_claimdef_* family of functions is now the vcx_credential_* family.
```
Comment (hadleym) [inline vcx/docs/migration.md]: Do we mean `is now the vcx_credentialdef_* family.` ... ?

File: vcx/docs/migration.md (hunk 1 above)
Comment (glowkey) [reply vcx/docs/migration.md]: Good catch, line 8 should actually say "vcx_credentialdef_*"

File: vcx/docs/migration.md (hunk 2 above)
Comment (glowkey) [reply vcx/docs/migration.md]: yes

**Pull request merged**